packages/vite/src/vite.ts

```diff
@@ -173,6 +173,12 @@ export async function bundle (nuxt: Nuxt) {
     })
   }
 
+  nuxt.hook('vite:extendConfig', (config) => {
+    if (config.publicDir) {
+      consola.warn('Using `options.vite.publicDir` option is not supported together with Nuxt. Use `options.dir.public` instead. You can read more in `https://nuxt.com/docs/api/configuration/nuxt-config#public`.')
+    }
+  })
```

Comment:
`if (config.publicDir) {...}` checks that `config.publicDir !== false && config.publicDir !== ''`, both of which prevent public assets from being copied.

This warning message is shown twice in Terminal, because it's run for client build and server build. Should I show it only in either of them by checking `isClient` or `isServer` props?

Reply:
I've moved the warning to schema so it will only warn once. what do you think?